src/providers/mod.rs

```diff
@@ -64,6 +67,21 @@ fn create_file(filename: &str) -> Result<File> {
     File::create(file_path).chain_err(|| format!("failed to create file {:?}", file_path))
 }
 
+/// Add a message to the journal logging SSH key additions; this
+/// will be used by at least Fedora CoreOS to display in the console
+/// if no ssh keys are present.
+fn write_ssh_key_journal_entry(log: logging::Priority, name: &str, path: &str) -> Result<()> {
```

Comment:
We don't want this function to fail the ssh-keys flows, thus we are not interested in this returning a result.

You can make this infallible, dropping the `Result` here and the `Ok` at the bottom.